cdap-app-templates/cdap-etl/hydrator-test/src/main/java/co/cask/cdap/etl/mock/batch/LookupTransform.java

```diff
@@ -65,6 +68,14 @@ public void initialize(TransformContext context) throws Exception {
   @Override
   public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter) throws Exception {
     T lookedUpValue = lookup.lookup((String) input.get(config.lookupKey));
+    R lookedUpValueBytes = lookup.lookup(((String) input.get(config.lookupKey)).getBytes());
+    if (lookedUpValue instanceof Result) {
+      assert lookedUpValueBytes instanceof Result;
```

Comment:
throw an exception instead of using assert. if the jvm is run without a specific flag, assertions won't do anything.